realm/realm-library/src/objectServer/java/io/realm/mongodb/sync/SyncSession.java

```diff
@@ -192,17 +194,19 @@ public URI getServerUrl() {
     }
 
     // This callback will happen on the thread running the Sync Client.
-    void notifySessionError(String nativeErrorCategory, int nativeErrorCode, String errorMessage) {
+    void notifySessionError(String nativeErrorCategory, int nativeErrorCode, String errorMessage, String clientResetPathInfo) {
         if (errorHandler == null) {
             return;
         }
         ErrorCode errCode = ErrorCode.fromNativeError(nativeErrorCategory, nativeErrorCode);
         if (errCode == ErrorCode.CLIENT_RESET) {
             // errorMessage contains the path to the backed up file
-            RealmConfiguration backupRealmConfiguration = configuration.forErrorRecovery(errorMessage);
-            clientResetHandler.onClientReset(this, new ClientResetRequiredError(appNativePointer, errCode, "A Client Reset is required. " +
-                    "Read more here: https://docs.realm.io/sync/using-synced-realms/errors#client-reset.",
-                    configuration, backupRealmConfiguration));
+            if (clientResetPathInfo == null) {
```

Comment:
What is the reason for keeping this? It should never be the case as we won't trigger the callback if the path is not present. If keeping it then maybe add a _should never happen..._-comment

Reply:
In the past there were two error handlers, one for sessions and one for the SyncClient. We routed both through the same SyncSession handler on the Java side as that was easier to consume. But the SyncClient error reporting was removed at some point so everything goes through the sessions, just like on the Java side. So this was just leftover code that we forgot to remove.